### ISSUE TITLE
Renaming attributes

### ISSUE THREAD

**Issue by dvalters** [dvalters/fuse-netcdf#27]
Current status:

In a Variable directory:
```
# With rename
[dvalters@helmsdale blh]$ rename newattr newattr_bar newattr
rename: newattr: rename to newattr_bar failed: Read-only file system

# Same for mv
[dvalters@helmsdale blh]$ mv newattr newattr_bar
mv: cannot move ‘newattr’ to ‘newattr_bar’: Read-only file system
```

Current debug (-vv) output:

```
getattr(u'/blh/newattr_bar', None)
getattr: /blh/newattr_bar does not exist
getattr(u'/blh/newattr', None)
getattr(u'/blh/newattr_bar', None)
getattr: /blh/newattr_bar does not exist
getattr(u'/blh/newattr_bar', None)
getattr: /blh/newattr_bar does not exist
rename(u'/blh/newattr', u'/blh/newattr_bar')

```

Needs an implementation of `rename()`!

**Comment by dvalters** [comment]
Works on empty attributes, but not non-empty ones?
```
[dav@localhost blh]$ ll
total 0
-rw-r--r--+ 1 dav dav      18 Aug  8 23:22 add_offset
-rw-r--r--+ 1 dav dav 8147569 Aug  8 23:22 DATA_REPR
-rw-r--r--+ 1 dav dav       7 Aug  8 23:22 _FillValue
-rw-r--r--+ 1 dav dav       0 Aug  8 23:22 fooroobar
-rw-r--r--+ 1 dav dav      22 Aug  8 23:22 long_name
-rw-r--r--+ 1 dav dav       7 Aug  8 23:22 missing_value
-rw-r--r--+ 1 dav dav       7 Aug  8 23:22 newattr
-rw-r--r--+ 1 dav dav      20 Aug  8 23:22 scale_factor
-rw-r--r--+ 1 dav dav       2 Aug  8 23:22 units
[dav@localhost blh]$ rename newattr newerattr newattr 
rename: newattr: rename to newerattr failed: Invalid argument
[dav@localhost blh]$ 
```

```
Uncaught exception from FUSE operation rename, returning errno.EINVAL.
Traceback (most recent call last):
  File "/home/dav/devel/fusepy/fuse.py", line 715, in _wrapper
    return func(*args, **kwargs) or 0
  File "/home/dav/devel/fusepy/fuse.py", line 787, in rename
    new.decode(self.encoding))
  File "/home/dav/devel/fusepy/fuse.py", line 1057, in __call__
    return getattr(self, op)(*args)
  File "fusenetcdf/fusenetcdf.py", line 387, in newfunc
    result = attr(*args, **kwargs)
  File "fusenetcdf/fusenetcdf.py", line 454, in rename
    return self.ncfs.rename(old, new)
  File "fusenetcdf/fusenetcdf.py", line 357, in rename
    self.rename_var_attr(old, new)
  File "fusenetcdf/fusenetcdf.py", line 232, in rename_var_attr
    cur_var.renameAttribute(old_attr_name, new_attr_name)
  File "netCDF4/_netCDF4.pyx", line 4027, in netCDF4._netCDF4.Variable.renameAttribute
  File "netCDF4/_netCDF4.pyx", line 1733, in netCDF4._netCDF4._ensure_nc_success
RuntimeError: NetCDF: Operation not allowed in data mode
getattr(u'/blh', None)

```

**Comment by dvalters** [comment]
See: https://www.unidata.ucar.edu/software/netcdf/workshops/2012/netcdf4_programs/DefineMode.html

**Comment by dvalters** [comment]
Working now, just need to make sure rename function also renames attributes correctly/does not create inconsistent dataset and possibly check `unlink()` function?

**Comment by dvalters** [comment]
Merged from #29 - still to check previous comment before closing

**Comment by dvalters** [comment]
Closed by #32
